### DISCUSSION TITLE
Variables: Multiple capture groups using regex

### DISCUSSION THREAD

**Discussion by souravsharma6633** [grafana/grafana#38255] [Dashboards, Panels & Visualizations]
**What would you like to be added**:
I am not sure if it is already there. I am having loki as datasource. My main problem is to capture multiple values from a single string having variable lengths.
Consider a,b,c as a label. Now I want to capture all 3 values (a b and c). For that, I am using label_values(label) and then using a simple regex as /(.*?),(.*?),(.*?)/.
But the result is 'a' only. Rather it should be 'a' 'b' 'c'.
It captures only first value in a regex having multiple capture groups.
Maybe its a bug or if feature is not there, its a very useful feature.

**Why is this needed**:
It helps in extracting multiple label values from a single string. We could even use this feature in usecase below-
Consider having multiple arrays ([a,b,c] , [e,f] , [m]) sent to loki as string labels (as arrays can't be used as labels). The arrays are of variable lengths. Using this feature, we could extract all label values and query it.

Consider it as a question also and reply with necessary info.

**Comment by ivanahuckova** [comment]
@hugohaggmark is this currently possible? I went trough the [documentation](https://grafana.com/docs/grafana/latest/variables/filter-variables-with-regex/#filter-variables-with-regex) and couldn't find example of this. If it is, we could add it to documentation. If not, then I would pass it to Platform Backlog. 🙂

**Comment by hugohaggmark** [comment]
@ivanahuckova only named group `text` and `value` can be used currently. A third unnamed/named capture group is not supported yet.

**Comment by pkolyvas** [comment]
We're in the process of moving Grafana feature requests to [GitHub Discussions](https://community.grafana.com/t/were-moving-grafana-feature-requests-to-github-discussions/50341/3). This issue is being moved to [Discussions](https://github.com/grafana/grafana/discussions) or has been moved from our issues list, depending on where you're seeing this message. 

 We hope to use the discussion thread to continue building use cases, and additional evidence based on engagement/voting, in order to help us prioritize this feature request later on down the road. 

As with all things Grafana, we value your input and participation in this project and appreciate your patience during this transitional period. Looking forward to the discussion!

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
